serdes/avro-serde/src/main/java/io/apicurio/registry/serde/avro/AvroKafkaDeserializer.java

```diff
@@ -45,55 +43,45 @@
 public class AvroKafkaDeserializer<U> extends AbstractKafkaDeserializer<Schema, U> {
 
     private final DecoderFactory decoderFactory = DecoderFactory.get();
+    private AvroSchemaParser parser = new AvroSchemaParser();
     private AvroDatumProvider<U> avroDatumProvider;
     private AvroEncoding configEncoding;
+    private AvroSerdeHeaders avroHeaders;
 
     public AvroKafkaDeserializer() {
-        this(null);
+        super();
     }
 
     public AvroKafkaDeserializer(RegistryClient client) {
-        this(client, new DefaultAvroDatumProvider<>());
-    }
-
-    public AvroKafkaDeserializer(RegistryClient client, AvroDatumProvider<U> avroDatumProvider) {
         super(client);
-        setAvroDatumProvider(avroDatumProvider);
     }
 
-    public AvroKafkaDeserializer<U> setAvroDatumProvider(AvroDatumProvider<U> avroDatumProvider) {
+    private AvroKafkaDeserializer<U> setAvroDatumProvider(AvroDatumProvider<U> avroDatumProvider) {
```

Comment:
Does this need to be public for cases where the deserializer is instantiated but not configured?  I think for connectors or perhaps kafka streams, the serde classes are directly instantiated rather than created via properties.

Reply:
For the connectors that is configured via properties, but I don't know about kafka streams.

Reply:
OK looking at the `org.apache.kafka.common.serialization.Serde<T>` interface, configuration is done via properties.  So let's keep this private and we can always change it back if someone hits a problem.